inventory.py

```diff
@@ -69,6 +76,79 @@ def slim_down_course_data(course_data: Sequence[Dict]) -> Sequence[Dict]:
         slim_course_dicts.append(slim_course_dict)
     return slim_course_dicts
 
+def get_next_page_url(response):
+    """
+    get the next page url from the Http response headers
+    :param response:
+    :type response: requests
+    :return: next_page_url
+    :rtype: str
+    """
+    logging.debug(get_next_page_url.__name__ + '() called')
+
+    if not response.links:
+        logging.debug('The api call do not have Link headers')
+        return None
+
+    for page in response.links:
+        if 'next' in page:
+            return response.links['next']['url']
+
+
+def handle_request_if_failed(response):
+    if response.status_code != 200:
+        return False
+    else:
+        return True
+
+
+def get_course_publish_unpublish_date(df_row: str, publish_or_unpublish_date, next_page_url = None)->str:
+    logger.debug(df_row)
+    if df_row['course_workflow_state'] == 'available':
+        event_type_to_check_for = 'published'
+    if df_row['course_workflow_state'] == 'unpublished':
+        event_type_to_check_for = 'claimed'
+
+    course_id= df_row['course_id']
+
+    if next_page_url is not None:
+        url = next_page_url
+    else:
+        url = f"{API_SCOPE_PREFIX}/audit/course/courses/{course_id}?per_page={PER_PAGE_COUNT}"
+
+    try:
+        response = make_audit_logs_api_request(url)
+
+    except (Exception) as e:
+        logger.exception('getting sections has erroneous response ' + e.message)
+        return publish_or_unpublish_date
+
+    if not handle_request_if_failed(response):
+        return publish_or_unpublish_date
+
+    audit_events = json.loads(response.text.encode('utf8'))
+    if not audit_events:
+        return publish_or_unpublish_date
+
+    events = audit_events['events']
+
+    for event in events:
+        if event['event_type'] == event_type_to_check_for:
+            publish_or_unpublish_date = event['created_at']
+            logger.info(f"Date for Workflow type :{df_row['course_workflow_state']} for course {course_id}")
+            break
+    if publish_or_unpublish_date is None:
+        next_page_url = get_next_page_url(response)
+        if next_page_url is not None:
+            get_course_publish_unpublish_date(course_id, publish_or_unpublish_date, next_page_url)
```

Comment:
How does this work here? The recursion idea is kinda cool, but I don't see you using the results from querying the next page?

Reply:
See My comments here https://github.com/tl-its-umich-edu/course-inventory/pull/9#issuecomment-600088985